parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriteStoreBase.java

```diff
@@ -237,10 +237,11 @@ private void sizeCheck() {
       } else {
         rowCountForNextRowCountCheck = min(rowCountForNextRowCountCheck, writer.getRowsWrittenSoFar() + pageRowCountLimit);
       }
+      //estimate remaining row count by previous input for next row count check
       long rowsToFillPage =
           usedMem == 0 ?
               props.getMaxRowCountForPageSizeCheck()
-              : (long) rows / usedMem * remainingMem;
+              : rows * remainingMem / usedMem;
```

Comment:
Should we worry about overflow by (rows * remainingMem)?

Reply:
I've felt it as a low risk. We are talking about pages here so the neither the row count nor the used memory should be too high that this would overflow.

Reply:
Same as gszadovszky +1